server/asset/eth/eth.go

```diff
@@ -242,34 +239,40 @@ var _ asset.AccountBalancer = (*ETHBackend)(nil)
 
 // unconnectedETH returns a Backend without a node. The node should be set
 // before use.
-func unconnectedETH(logger dex.Logger, net dex.Network) (*ETHBackend, error) {
+func unconnectedETH(bipID uint32, contractAddr common.Address, vTokens map[uint32]*VersionedToken, logger dex.Logger, net dex.Network) (*ETHBackend, error) {
 	// TODO: At some point multiple contracts will need to be used, at
 	// least for transitory periods when updating the contract, and
 	// possibly a random contract setup, and so this section will need to
 	// change to support multiple contracts.
-	contractAddr, exists := dexeth.ContractAddresses[0][net]
-	if !exists || contractAddr == (common.Address{}) {
-		return nil, fmt.Errorf("no eth contract for version 0, net %s", net)
-	}
 	return &ETHBackend{&AssetBackend{
 		baseBackend: &baseBackend{
-			net:        net,
-			baseLogger: logger,
-			tokens:     make(map[uint32]*TokenBackend),
+			net:             net,
+			baseLogger:      logger,
+			tokens:          make(map[uint32]*TokenBackend),
+			baseChainID:     bipID,
```

Comment:
Many of the error messages in this file and rpcclient.go confused me with references to "eth" or "ethereum".  Maybe like this: https://github.com/chappjc/dcrdex/commit/69a03e11a270456ef9bc3f26b2ea6a62ca41fa37